### PR TITLE
Plots limit lines

### PR DESCRIPTION
**Pull request by shefalijoshi**
Adds `limitDefinition` and `getLimits` methods to the Telemetry API which are hooks into the limit providers to obtain limit definitions for telemetry endpoints.
Limits returned by the limit providers are expected to have the following format:
```
{
	level1: {
		low: {
			key1: value1,
			key2: value2
		},
		high: {
			key1: value1,
			key2: value2
		}
	},
	level2: {
		low: {
			key1: value1,
			key2: value2
		},
		high: {
			key1: value1,
			key2: value2
		}
	}
}
```
Resolves: #3884
May address: #2662, #526

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y
5. Testing instructions included? Y

### CODE REVIEW

File: src/plugins/plot/chart/LimitLabel.vue [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    computed: {
+        styleObj() {
+            const top = `${this.point.top - 10}px`;
+            const left = `${this.point.left + 5}px`;
+
+            return {
+                'position': 'absolute',
+                'top': top,
+                'left': left,
+                color: '#fff'
```
Comment (nikhilmandlik) [inline src/plugins/plot/chart/LimitLabel.vue:41]: 'color'?

File: src/plugins/plot/chart/LimitLine.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    :class="cssWithoutUprLwr"
+>
+</template>
+
+<script>
+export default {
+    props: {
+        point: {
+            type: Object,
+            required: true
```
Comment (nikhilmandlik) [inline src/plugins/plot/chart/LimitLine.vue:12]: I think, `required: true` need a default return?

Comment (shefalijoshi) [reply src/plugins/plot/chart/LimitLine.vue:12]: Done.

File: src/plugins/plot/chart/LimitLabel.vue (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/plot/chart/LimitLabel.vue:41]: Done

**Review by nikhilmandlik — APPROVED** [review]
LGTM

File: src/api/telemetry/TelemetryAPI.js [hunk 3]
```diff
@@ -504,6 +504,26 @@ define([
...
+     * If a provider does not implement this method, it is presumed
+     * that no limits are defined for this domain object's telemetry.
+     *
+     * @param {module:openmct.DomainObject} domainObject the domain
+     *        object for which to get limits
+     * @returns {module:openmct.TelemetryAPI~LimitEvaluator}
+     * @method limits
+     * @memberof module:openmct.TelemetryAPI~TelemetryProvider#
+     */
+    TelemetryAPI.prototype.limitDefinition = function (domainObject) {
```
Comment (akhenry) [inline src/api/telemetry/TelemetryAPI.js:523]: What is the purpose of this method? It just calls `getLimits`. This is public API and it needs to be really unambiguous.

File: src/api/telemetry/TelemetryAPI.js [hunk 4]
```diff
@@ -531,5 +551,42 @@ define([
         return provider.getLimitEvaluator(domainObject);
     };
 
+    /**
```
Comment (akhenry) [inline src/api/telemetry/TelemetryAPI.js:554]: Good documentation!

**Review by akhenry — COMMENTED** [review]
I like the API changes. There is just one ambiguity I'd like to address, and possible one place where we could save developers a step.

Thanks!

**Pull request merged**